pkg/datalayer/datalayer.go

```diff
@@ -1,28 +1,172 @@
 package datalayer
 
 import (
+	"encoding/json"
+	"fmt"
 	"os"
+	"os/exec"
 	"path/filepath"
+	"strings"
+
+	"github.com/nu7hatch/gouuid"
 )
 
 // ClusterHQ data layer, naive vfs (directory-based) implementation
 
 type DataLayer struct {
-	BasePath string
+	basePath string
+}
+
+type CommitId string
+type CommitMessage string
+
+type Commit struct {
+	Id      CommitId      `json:"id"`
+	Message CommitMessage `json:"message"`
+}
+
+func NewDataLayer(basePath string) *DataLayer {
+	return &DataLayer{basePath}
+}
+
+func (dl *DataLayer) volumePath(volumeName string) string {
+	return filepath.FromSlash(dl.basePath + "/" + volumeName)
+}
+
+func (dl *DataLayer) variantPath(volumeName, variantName string) string {
+	return filepath.FromSlash(dl.basePath + "/" + volumeName + "/branches/" + variantName)
+}
+
+func (dl *DataLayer) commitPath(volumeName string, commitId CommitId) string {
+	return filepath.FromSlash(dl.basePath + "/" + volumeName + "/commits/" + string(commitId))
```

Comment:
Ah sorry for the noise, I tested this with an example program and it built. I probably didn't recreate the problem correctly.